### PR TITLE
Replace IndicesQueriesRegistry

### PR DESCRIPTION
**Pull request by nik9000**
Replaces `IndicesQueriesRegistry` with` XContentParser#namedObject`.

This is the first of the real payoff for https://github.com/elastic/elasticsearch/pull/22003, one less thing to pass around the entire application. Now the parsers it contained are wrapped in the `XContentParser`.

### CODE REVIEW

**Comment by nik9000** [comment]
@rjernst or @imotov or @javanna or @martijnvg want to have a look? I'm not picky who looks....

File: core/src/main/java/org/elasticsearch/rest/action/RestActions.java [hunk 1]
```diff
@@ -195,9 +194,9 @@ public static QueryBuilder urlParamsToQueryBuilder(RestRequest request) {
         return queryBuilder;
     }
 
-    public static QueryBuilder getQueryContent(XContentParser requestParser, IndicesQueriesRegistry indicesQueriesRegistry,
-                                               ParseFieldMatcher parseFieldMatcher) {
-        QueryParseContext context = new QueryParseContext(indicesQueriesRegistry, requestParser, parseFieldMatcher);
+    public static QueryBuilder getQueryContent(XContentParser requestParser, ParseFieldMatcher parseFieldMatcher) {
+        // NOCOMMIT it is very close to time to remove this
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/rest/action/RestActions.java]: But not yet time, I think. This PR is big enough as is.

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
Thanks for reviewing @martijnvg! One more parser registry down and one less thing bound in guice. Hurray!

master: 567c65b0d57e92cd530dafa7f55de7ba681af13e
5.x: backporting

**Pull request merged**